packages/server/utils/MattermostServerManager.ts

```diff
@@ -0,0 +1,70 @@
+import fetch from 'node-fetch'
+
+// Mattermost is a server-only integration for now, unlike the Slack integration
+
+// We're following a similar manager pattern here should we wish to refactor the
+// Mattermost integration. We might want to do this if we decide to use Mattermost's
+// upcoming "App" framework
+
+export interface MattermostApiResponse {
+  ok: boolean
+  status: number
+  error: string | undefined
+}
+
+abstract class MattermostManager {
+  webhookUrl: string
+  abstract fetch
+  headers: any
+
+  constructor(webhookUrl: string) {
+    this.webhookUrl = webhookUrl
+  }
+
+  private async post(payload: any): Promise<MattermostApiResponse> {
+    const res = await fetch(this.webhookUrl, {
```

Comment:
this is my bad, we need to refactor the slack manager interface.
The best example of an interface we have today is AtlassianManager.
2 things to note:
- It has a `fetchWithTimeout` so if the request takes 2 minutes it won't bork our server (like Google did :cry: )
- we return an `Error` instance based on the gateway or payload error provided (ie the `ok` prop). This great because then whenever you call `post` you can look at the return value, check it's an error, & if so,  handle it & then you have a great strongly typed payload after that. Example usage here: https://github.com/ParabolInc/parabol/blob/31cc0e270ad29dd9b5a32fd55899eb6362426b09/packages/server/dataloader/atlassianLoaders.ts#L114-L118
- For your consideration: https://github.com/ParabolInc/parabol/blob/295c8fb078878d42a28c4d84bca4834b4d0d29a8/packages/client/utils/AtlassianManager.ts

Reply:
This is real, real nice. This `fetchWithTimeout` (and associating typing) definitely belongs on an abstract class. I'll lightly refactor what I have now and then pull this same pattern into the abstract integration manager class I'm working on